File: backend/src/main/java/com/bidbridge/backend/entity/Incoterm.java

package com.bidbridge.backend.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;
import java.time.Instant;

@Data
@NoArgsConstructor
@Entity
@Table(name = "incoterms", schema = "public")
public class Incoterm {

    @Id
    @GeneratedValue
    private UUID id;

    private String name;

    @ManyToOne
    @JoinColumn(name = "company_id")
    private Company company;

    private Instant createdAt;
}

File: frontend/src/types/incoterm.ts
export interface Incoterm {
  id: string;
  name: string;
  company_id: string;
  company_name: string;
  created_at: string;
}

export interface IncotermApiResponse {
  id: string;
  name: string;
  companyId: string;
  companyName: string;
  createdAt: string;
}

export function transformIncotermFromApi(
  apiIncoterm: IncotermApiResponse
): Incoterm {
  return {
    id: apiIncoterm.id,
    name: apiIncoterm.name,
    company_id: apiIncoterm.companyId,
    company_name: apiIncoterm.companyName,
    created_at: apiIncoterm.createdAt,
  };
}
